processor/src/decoder/mod.rs

```diff
@@ -370,6 +370,13 @@ impl Decoder {
         }
     }
 
+    /// Sets the helper registers in the trace to the user-provided helper values. This is expected
+    /// to be called during the execution of a user operation.
+    pub fn set_user_op_helpers(&mut self, op: Operation, values: &[Felt]) {
+        debug_assert!(!op.is_control_op(), "op is a control operation");
```

Comment:
I wonder if it would be better to get the `op` value from the decoder trace (rather than as an input parameter). Probably not something that should be a part of this PR - but maybe leave a TODO here? And maybe create a "good first issue" for this?

Reply:
I actually started out with the approach you suggest, but the way I've done it ended up being simpler than reconstructing the opcode from the trace, partly because checking if it's a control flow operation becomes a bit more complicated when we only have the opcode. I may have overlooked something obvious, but if not, then this might be a bit more involved than we want for a "good first issue"

Reply:
An alternative would be to change the DecoderTrace to have the last executed operation, and maybe that's the approach you intend from your comment? I considered that as well, but ultimately decided to hold off on modifying the DecoderTrace struct. That change would be simpler though (and would then be a "good first issue")

Reply:
I didn't really think through the potential approaches too deeply. What you described would work - though, we'd need this field only for debug purposes.

Another option could be to just compare opcodes of control flow operations one-by-one to the values in `op_bits` columns and error out if none match. This would be rather inefficient but since we are doing this in debug mode only, might be OK.

Reply:
I've opened #240

Reply:
Thank you!